### PR TITLE
pid_t: unify usage of special task IDs

### PR DESCRIPTION
**Pull request by pkarashchenko**
## Summary
Unify usage of special task IDs

## Impact
None. Refactoring only

## Testing
Pass CI

### CODE REVIEW

File: include/nuttx/sched.h [hunk 1]
```diff
@@ -79,7 +79,7 @@
 
 /* Special task IDS.  Any negative PID is invalid. */
 
-#define NULL_TASK_PROCESS_ID       (pid_t)0
```
Comment (xiaoxiang781216) [inline include/nuttx/sched.h:82]: should we change to IDLE_PROCESS_ID to algin with INVALID_PROCESS_ID?

File: drivers/syslog/syslog_device.c [hunk 2]
```diff
@@ -768,9 +768,10 @@ void syslog_dev_uninitialize(FAR struct syslog_channel_s *channel)
    * interrupt context.
    */
 
-  if (up_interrupt_context() || getpid() == 0)
+  if (up_interrupt_context() || getpid() == IDLE_TASK_PROCESS_ID)
```
Comment (xiaoxiang781216) [inline drivers/syslog/syslog_device.c:771]: should call sched_idle_task here

File: drivers/syslog/syslog_device.c [hunk 3]
```diff
@@ -768,9 +768,10 @@ void syslog_dev_uninitialize(FAR struct syslog_channel_s *channel)
...
     {
-      DEBUGASSERT(!up_interrupt_context() && getpid() != 0);
+      DEBUGASSERT(!up_interrupt_context() &&
+                  getpid() != IDLE_TASK_PROCESS_ID);
```
Comment (xiaoxiang781216) [inline drivers/syslog/syslog_device.c:774]: ditto

File: drivers/syslog/syslog_device.c [hunk 4]
```diff
@@ -321,7 +321,7 @@ static int syslog_dev_outputready(FAR struct syslog_dev_s *syslog_dev)
 
   /* Cases (4) and (5) */
 
-  if (up_interrupt_context() || getpid() == 0)
+  if (up_interrupt_context() || getpid() == IDLE_TASK_PROCESS_ID)
```
Comment (xiaoxiang781216) [inline drivers/syslog/syslog_device.c:324]: ditto

File: drivers/syslog/syslog_device.c (hunk 2 above)
Comment (xiaoxiang781216) [reply drivers/syslog/syslog_device.c:771]: sched_idle_task->sched_idletask

**Review by xiaoxiang781216 — APPROVED** [review]

File: arch/ceva/src/common/up_assert.c [hunk 5]
```diff
@@ -296,7 +296,7 @@ static void _up_assert(int errorcode)
 
   /* Are we in an interrupt handler or the idle task? */
 
-  if (CURRENT_REGS || running_task()->pid == 0)
+  if (up_interrupt_context() || sched_idletask())
```
Comment (pkarashchenko) [inline arch/ceva/src/common/up_assert.c:299]: @xiaoxiang781216 please review this place. Change seems to be equivalent, but maybe there is something that I miss.

File: sched/task/task_posixspawn.c [hunk 6]
```diff
@@ -410,7 +410,7 @@ int posix_spawn(FAR pid_t *pid, FAR const char *path,
    * for use within the OS.
    */
 
-  ret = nx_waitpid(proxy, &status, 0);
+  ret = nx_waitpid((pid_t)proxy, &status, 0);
```
Comment (gustavonihei) [inline sched/task/task_posixspawn.c:413]: How about keeping `proxy` as `pid_t` and instead receive the return from `kthread_create` with the `ret` local?

Comment (pkarashchenko) [reply sched/task/task_posixspawn.c:413]: Yes. This can be an option. I will implement the change

**Review by gustavonihei — APPROVED** [review]
Just made one comment for improving consistency, but overall LGTM.

**Comment by gustavonihei** [comment]
By the way, nice work @pkarashchenko!
It is a pleasure to see these inconsistencies being fixed throughout the NuttX code base.

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**